src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -25,94 +53,1321 @@
+                    teamMembersWithResponses.clear(); 
+                }
+                
+                teamsWithResponses.add(currentTeam);
+                if (!isChangeOfSection) {
+                    sectionPanel.getIsTeamWithResponses().put(currentTeam, true);
+                }
+            }
+            
+            // Change in section
+            if (isChangeOfSection) {
+                boolean isFirstSection = sectionPanel.getParticipantPanels().isEmpty();
+                if (!isFirstSection) {
+                    // Finalize building of section panel,
+                    finaliseBuildingSectionPanel(sectionPanel, prevSection, responsesGroupedByTeam,
+                                                 teamsWithResponses);
+                    
+                    // add sectionPanel to sectionPanels
+                    buildMissingTeamAndParticipantPanelsWithModerationButtonForSection(sectionPanel, prevSection, 
+                                                    teamsWithResponses);
+                    sectionPanels.put(prevSection, sectionPanel);
+                    sectionsWithResponses.add(prevSection);
+                    
+                    // and set up for next section
+                    teamsWithResponses.clear();
+                    teamsWithResponses.add(currentTeam);
+                    
+                    sectionPanel = new InstructorFeedbackResultsSectionPanel();
+                }
+                
+                sectionPanel.getIsTeamWithResponses().put(currentTeam, true);
+            }
+            
+            // For the current giver, constructs a panel 
+            InstructorFeedbackResultsGroupByQuestionPanel giverPanel 
+                    = buildParticipantGroupByQuestionPanel(giverIdentifier, responsesFromGiver,
+                                                           additionalInfoId, giverIndex);
+            
+            // add constructed InstructorFeedbackResultsGroupByQuestionPanel into section's participantPanels            
+            addParticipantPanelToSectionPanel(sectionPanel, currentTeam, giverPanel);
+            teamMembersWithResponses.add(giverIdentifier);
+            
+            prevTeam = currentTeam;
+            prevSection = currentSection;
+        }
+        
+        buildMissingParticipantPanelsForTeam(sectionPanel, prevTeam, teamMembersWithResponses, viewType.isFirstGroupedByGiver());
+        
+        teamsWithResponses.add(prevTeam);
+        buildMissingTeamAndParticipantPanelsForSection(sectionPanel, prevSection, teamsWithResponses, viewType.isFirstGroupedByGiver());
+        
+        finaliseBuildingSectionPanel(sectionPanel, prevSection, responsesGroupedByTeam, teamsWithResponses);
+        sectionPanels.put(prevSection, sectionPanel);
+
+        // TODO introduce enums for this, because this causes problems if there is a section named "All"
+        if (selectedSection.equals("All")) {
```

Comment:
May be good to replace all of these with `isAllSectionsSelected()` for now, we can change the boolean condition by editing that method alone later that way.